hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -491,7 +492,7 @@ public Pair<HoodieMetadataMergedLogRecordReader, Long> getLogRecordScanner(List<
         .withFileSystem(metadataMetaClient.getFs())
         .withBasePath(metadataBasePath)
         .withLogFilePaths(sortedLogFilePaths)
-        .withReaderSchema(schema)
+        .withReaderSchema(AvroInternalSchemaConverter.convertToEmpty(schema))
```

Comment:
HoodieBackedTableMetadata no support scheam evolution.
no need to modfiy those codes.

Reply:
I was asked to remove the second method https://github.com/apache/hudi/pull/5830#discussion_r1018637325